modules/jooby-kafka/src/main/java/io/jooby/kafka/KafkaModule.java

```diff
@@ -0,0 +1,156 @@
+   *  kafka.producer.value.serializer = "org.apache.kafka.common.serialization.StringSerializer"
+   * }</pre>
+   *
+   * Creates a new kafka consumer module using the <code>kafka.consumer</code> property key.
+   * This key must be present in the application configuration file, like:
+   *
+   * <pre>{@code
+   *  kafka.consumer.bootstrap.servers = "localhost:9092"
+   *  kafka.consumer.group.id = "group A"
+   *  kafka.consumer.key.deserializer = "org.apache.kafka.common.serialization.StringDeserializer"
+   *  kafka.consumer.value.deserializer = "org.apache.kafka.common.serialization.StringDeserializer"
+   * }</pre>
+   */
+  public KafkaModule() {
+    this("kafka.producer", "kafka.consumer");
+  }
+
+  /**
+   * Creates a new kafka producer module. The producer parameter can be one of:
+   *
+   * - A property key defined in your application configuration file, like <code>producerKey</code>.
+   *
+   * @param producerKey Database key
+   *
+   * Creates a new kafka consumer module. The consumer parameter can be one of:
+   *
+   * - A property key defined in your application configuration file, like <code>consumerKey</code>.
+   *
+   * @param consumerKey Database key
+   */
+  public KafkaModule(@Nonnull String producerKey, @Nonnull String consumerKey) {
+    this.producerKey = producerKey;
+    this.consumerKey = consumerKey;
+  }
+
+  /**
+   * Creates a new kafka module.
+   *
+   * @param producerProps kafka producer properties.
+   * @param consumerProps kafka consumer properties.
+   */
+  public KafkaModule(@Nonnull Properties producerProps, @Nonnull Properties consumerProps) {
+    this("kafka.producer", "kafka.consumer");
+    this.producerProps = producerProps;
+    this.consumerProps = consumerProps;
+  }
+
+
+  @Override
+  public void install(@Nonnull Jooby application) {
+    Config config = application.getConfig();
+
+    if (this.producerProps == null) {
+      this.producerProps = properties(config, this.producerKey);
+    }
+
+    if (this.consumerProps == null) {
+      this.consumerProps = properties(config, this.consumerKey);
+    }
+
+    ServiceRegistry registry = application.getServices();
+
+    if (this.producerProps != null) {
+      KafkaProducer producer = new KafkaProducer(producerProps);
+      application.onStop(producer::close);
+      registry.putIfAbsent(KafkaProducer.class, producer);
+    }
+
+    if (this.consumerProps != null) {
+      KafkaConsumer consumer = new KafkaConsumer(consumerProps);
+      application.onStop(consumer::close);
+      registry.putIfAbsent(KafkaConsumer.class, consumer);
+    }
+  }
+
+  private static Properties properties(final Config config, final String key) {
+    Properties props = new Properties();
+    if (config.hasPath(key)) {
+      // dump
+      config.getConfig(key).entrySet().forEach(
+          e -> props.setProperty(e.getKey(), e.getValue().unwrapped().toString()));
+    }
```

Comment:
You could throw an exception in an `else` block here, since this also means that configuration properties are missing.